### PR TITLE
Excludes clockwork wargolems from golem mutation toxin, nerfs normal clockwork golems

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
fix: You can no longer become an enhanced clockwork golem with mutation toxin.
balance: Normal clockwork golems now have 20% armor, down from 40%.
/:cl:

This was absolutely an oversight! War clockgolems are meant to be given to servants only since they're meant to be "human 2.0" - the main drawback to their existence is the fact that they're meant to be always valid, and that deftly circumvented it. You can no longer become that golem subtype through mutation.

Normal clockwork golems remain, however; they still have the restrictions of normal golems (no clothing etc.) but keep their other attributes like speed. Their innate armor has been dropped from 40% (which was already 15% less than normal golem armor) to 20% to compensate for that fact.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 1]
```diff
@@ -47,7 +48,11 @@
 
 /datum/species/golem/random/on_species_gain(mob/living/carbon/C, datum/species/old_species)
 	..()
-	var/list/golem_types = typesof(/datum/species/golem) - src.type
+	var/list/golem_types = subtypesof(/datum/species/golem)
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: this never changes, make it a static list init at round start

use a subsystem of some appropriate flavour to set it up and remove non valid types

**Review by optimumtact — CHANGES_REQUESTED** [review]

**Comment by AdamElTablawy** [comment]
And so, sand golems and runic golems remain completely superior to brass golems.

File: code/modules/mob/living/carbon/human/species_types/golems.dm (hunk 1 above)
Comment (Xhuis) [reply code/modules/mob/living/carbon/human/species_types/golems.dm]: Would assigning it in the first `on_species_gain` work as well?

Comment (optimumtact) [reply code/modules/mob/living/carbon/human/species_types/golems.dm]: sure

Comment (AnturK) [reply code/modules/mob/living/carbon/human/species_types/golems.dm]: Please do the cache

Comment (Xhuis) [reply code/modules/mob/living/carbon/human/species_types/golems.dm]: Sorry, I've been sick and it slipped my mind. I'll get this done when I get a chance.

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 2]
```diff
@@ -44,11 +45,20 @@
...
 	..()
-	var/list/golem_types = typesof(/datum/species/golem) - src.type
-	var/datum/species/golem/golem_type = pick(golem_types)
+	if(!random_golem_types)
+		random_golem_types = subtypesof(/datum/species/golem) - type
+		for(var/V in random_golem_types)
+			var/datum/species/golem/G = V
+			if(!initial(G.random_eligible))
+				random_golem_types -= G
+				to_chat(world, "Excluding golem type [initial(G.id)]")
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: @AnturK :grin:

Comment (Xhuis) [reply code/modules/mob/living/carbon/human/species_types/golems.dm]: Feature

Comment (AnturK) [reply code/modules/mob/living/carbon/human/species_types/golems.dm]: 🙈

**Pull request merged**